libwild/src/symbol_db.rs

```diff
@@ -596,45 +598,65 @@ pub(crate) fn resolve_alternative_symbol_definitions<'data>(
 
-    let replacements: Vec<Vec<(SymbolId, SymbolId)>> = buckets
+    let replacements: Vec<(SymbolId, SymbolId)> = buckets
         .par_iter_mut()
-        .map(|bucket| {
-            let mut replacements = Vec::new();
-
-            let mut add_replacement = |a, b| {
-                replacements.push((a, b));
-            };
+        .flat_map_iter(|bucket| {
+            let mut local_replacements = Vec::new();
 
             for (first, alternatives) in replace(
                 &mut bucket.alternative_definitions,
                 HashMap::with_hasher(RandomState::new()),
             ) {
-                let selected = select_symbol(symbol_db, first, &alternatives, resolved);
-                add_replacement(first, selected);
-
-                for alt in alternatives {
-                    add_replacement(alt, selected);
+                match select_symbol(symbol_db, first, &alternatives, resolved) {
+                    Ok(selected) => {
+                        local_replacements.push((first, selected));
+                        for alt in alternatives {
+                            local_replacements.push((alt, selected));
+                        }
+                    }
+                    Err(err) => {
+                        error_queue.push(err);
+                    }
                 }
             }
 
             for (first, alternatives) in replace(
                 &mut bucket.alternative_versioned_definitions,
                 HashMap::with_hasher(RandomState::new()),
             ) {
-                let selected = select_symbol(symbol_db, first, &alternatives, resolved);
-                add_replacement(first, selected);
-
-                for alt in alternatives {
-                    add_replacement(alt, selected);
+                match select_symbol(symbol_db, first, &alternatives, resolved) {
+                    Ok(selected) => {
+                        local_replacements.push((first, selected));
+                        for alt in alternatives {
+                            local_replacements.push((alt, selected));
+                        }
+                    }
+                    Err(err) => {
+                        error_queue.push(err);
+                    }
                 }
             }
-
-            replacements
+            local_replacements.into_iter()
         })
         .collect();
 
-    for (a, b) in replacements.iter().flat_map(|r| r.iter()) {
+    let mut duplicate_errors: Vec<anyhow::Error> = error_queue.into_iter().collect();
+    duplicate_errors.sort_by_key(|e| e.to_string());
+
+    if !duplicate_errors.is_empty() {
+        let error_details = duplicate_errors
+            .iter()
+            .map(|e| e.to_string())
+            .collect::<Vec<_>>()
+            .join("\n");
+        return Err(anyhow::Error::msg(format!(
+            "Duplicate symbols detected: {error_details}"
+        )));
+    }
+
+    for (a, b) in &replacements {
```

Comment:
So with your most recent push, it's still collecting all the replacements into one large Vec - it's just doing it in a slightly different way. What I was meaning was change this line back to:

```rust
for (a, b) in replacements.iter().flat_map(|r| r.iter()) {
```

Then the `Vec` that gets collected above contains many smaller Vecs like it did before, which means less copying.